libs/designsystem/menu/src/menu.component.ts

```diff
@@ -75,33 +83,261 @@ export class MenuComponent implements AfterViewInit, OnDestroy {
+        this.floatingMenu.show();
+        this.focusItem();
+        this.preventDefaultAndStopImmediatePropagation(event);
+        break;
+    }
+  }
+
+  private isPrintableCharacter(key: string) {
+    return key.length === 1 && key.match(/\S/);
+  }
+
+  private handleKeyDownForOpenedMenu(event: KeyboardEvent) {
+    const key = event.key;
+
+    switch (key) {
+      case 'ArrowDown':
+        if (this.focusedIndex === this.kirbyItems.length - 1) {
+          this.focusedIndex = 0;
+        } else {
+          this.focusedIndex++;
+        }
+        this.focusItem();
+        this.preventDefaultAndStopImmediatePropagation(event);
+        break;
+      case 'ArrowUp':
+        if (this.focusedIndex === 0) {
+          this.focusedIndex = this.kirbyItems.length - 1;
+        } else {
+          this.focusedIndex--;
+        }
+        this.focusItem();
+        this.preventDefaultAndStopImmediatePropagation(event);
+        break;
+      case 'Home': {
+        if (this.focusedIndex > 0) {
+          this.focusedIndex = 0;
+          this.focusItem();
+        }
+        this.preventDefaultAndStopImmediatePropagation(event);
+        break;
+      }
+      case 'End': {
+        if (this.focusedIndex < this.kirbyItems.length - 1) {
+          this.focusedIndex = this.kirbyItems.length - 1;
+          this.focusItem();
+        }
+        this.preventDefaultAndStopImmediatePropagation(event);
+        break;
+      }
+      case 'Escape':
+        if (this.closeOnEscapeKey) {
+          this.floatingMenu.hide();
+        }
+        this.preventDefaultAndStopImmediatePropagation(event);
+        break;
+      case 'Tab':
+        this.floatingMenu.hide();
+        break;
+      default: {
+        if (this.isPrintableCharacter(key)) {
+          const foundItemIndex = this.getIndexOfItemByFirstCharacter(key);
+          if (foundItemIndex > -1) {
+            this.focusedIndex = foundItemIndex;
+            this.focusItem();
+          }
+          this.preventDefaultAndStopImmediatePropagation(event);
+        }
+      }
+    }
+  }
+
+  private getIndexOfItemByFirstCharacter(char: string) {
+    return this.getIndexByFirstMatchingStartString(
+      char,
+      this.kirbyItems.map((item) => item.nativeElement.innerText),
+      this.focusedIndex + 1
+    );
+  }
+
+  private getIndexByFirstMatchingStartString(
+    matchString: string,
+    words: string[],
+    startIndex: number
+  ): number {
+    matchString = matchString.toLowerCase();
+
+    const wordsStartingWithMatchString = words
+      .map((word, index) => {
+        return { word: word.toLowerCase(), index };
+      })
+      .filter((match) => match.word.startsWith(matchString));
+
+    if (wordsStartingWithMatchString.length === 0) {
+      return -1;
+    }
+
+    const firstWordStartingWithChar = wordsStartingWithMatchString[0];
+    const nextWordStartingWithChar = wordsStartingWithMatchString.filter(
+      (wordAndIndex) => wordAndIndex.index >= startIndex
+    )[0];
```

Comment:
Consider using builtin function to find first match:
```suggestion
    const nextWordStartingWithChar = wordsStartingWithMatchString.find(
      (wordAndIndex) => wordAndIndex.index >= startIndex
    );
```